platform/linux-generic/pktio/dpdk.c

```diff
@@ -29,14 +29,27 @@
 
 #include <rte_config.h>
 #include <rte_malloc.h>
+#if __GNUC__ >= 7
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wimplicit-fallthrough=0"
+#endif
 #include <rte_mbuf.h>
+#if __GNUC__ >= 7
+#pragma GCC diagnostic pop
+#endif
 #include <rte_mempool.h>
 #include <rte_ethdev.h>
 #include <rte_ip.h>
 #include <rte_ip_frag.h>
+#include <rte_log.h>
 #include <rte_udp.h>
 #include <rte_tcp.h>
 #include <rte_string_fns.h>
+#include <rte_version.h>
+
+#if RTE_VERSION < RTE_VERSION_NUM(17, 8, 0, 0)
+#define rte_log_set_global_level rte_set_log_level
+#endif
```

Comment:
This breaks build on Ubuntu 17.10, which uses DPDK 17.05: 'error: ‘rte_set_log_level’ is deprecated' 

So the correct version to use here is RTE_VERSION_NUM(17, 5, 0, 0)